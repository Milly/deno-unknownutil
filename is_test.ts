import { assertEquals } from "./deps_test.ts";
import {
  isArray,
  isFunction,
  isNone,
  isNull,
  isNumber,
  isObject,
  isString,
  isUndefined,
} from "./is.ts";

Deno.test("isString returns true on array", () => {
  assertEquals(isString(""), true);
  assertEquals(isString("Hello World"), true);
});
Deno.test("isString returns false on non array", () => {
  assertEquals(isString(0), false);
  assertEquals(isString([]), false);
  assertEquals(isString({}), false);
  assertEquals(isString(function () {}), false);
  assertEquals(isString(null), false);
  assertEquals(isString(undefined), false);
});

Deno.test("isNumber returns true on array", () => {
  assertEquals(isNumber(0), true);
  assertEquals(isNumber(1234567890), true);
});
Deno.test("isNumber returns false on non array", () => {
  assertEquals(isNumber(""), false);
  assertEquals(isNumber([]), false);
  assertEquals(isNumber({}), false);
  assertEquals(isNumber(function () {}), false);
  assertEquals(isNumber(null), false);
  assertEquals(isNumber(undefined), false);
});

Deno.test("isArray returns true on array", () => {
  assertEquals(isArray([]), true);
  assertEquals(isArray([0, 1, 2]), true);
  assertEquals(isArray(["a", "b", "c"]), true);
  assertEquals(isArray([0, "a", 1]), true);
});
Deno.test("isArray returns false on non array", () => {
  assertEquals(isArray(""), false);
  assertEquals(isArray(0), false);
  assertEquals(isArray({}), false);
  assertEquals(isArray(function () {}), false);
  assertEquals(isArray(null), false);
  assertEquals(isArray(undefined), false);
});
Deno.test("isArray<T> returns true on T array", () => {
  assertEquals(isArray([0, 1, 2], isNumber), true);
  assertEquals(isArray(["a", "b", "c"], isString), true);
});
Deno.test("isArray<T> returns false on non T array", () => {
  assertEquals(isArray([0, 1, 2], isString), false);
  assertEquals(isArray(["a", "b", "c"], isNumber), false);
});

Deno.test("isObject returns true on object", () => {
  assertEquals(isObject({}), true);
  assertEquals(isObject({ a: 0 }), true);
  assertEquals(isObject({ a: "a" }), true);
});
Deno.test("isObject returns false on non object", () => {
  assertEquals(isObject(""), false);
  assertEquals(isObject(0), false);
  assertEquals(isObject([]), false);
  assertEquals(isObject(function () {}), false);
  assertEquals(isObject(null), false);
  assertEquals(isObject(undefined), false);
});
Deno.test("isObject<T> returns true on T object", () => {
  assertEquals(isObject({ a: 0 }, isNumber), true);
  assertEquals(isObject({ a: "a" }, isString), true);
});
Deno.test("isObject<T> returns false on non T object", () => {
  assertEquals(isObject({ a: 0 }, isString), false);
  assertEquals(isObject({ a: "a" }, isNumber), false);
});

Deno.test("isFunction returns true on function", () => {
  assertEquals(isFunction(isFunction), true);
  assertEquals(isFunction(function () {}), true);
  assertEquals(isFunction(() => {}), true);
  assertEquals(isFunction(setTimeout), true);
});
Deno.test("isFunction returns false on non function", () => {
  assertEquals(isFunction(""), false);
  assertEquals(isFunction(0), false);
  assertEquals(isFunction([]), false);
  assertEquals(isFunction({}), false);
  assertEquals(isFunction(null), false);
  assertEquals(isFunction(undefined), false);
});

Deno.test("isNull returns true on null", () => {
  assertEquals(isNull(null), true);
});
Deno.test("isNull returns false on non null", () => {
  assertEquals(isNull(""), false);
  assertEquals(isNull(0), false);
  assertEquals(isNull([]), false);
  assertEquals(isNull({}), false);
  assertEquals(isNull(function () {}), false);
  assertEquals(isNull(undefined), false);
});

Deno.test("isUndefined returns true on null", () => {
  assertEquals(isUndefined(undefined), true);
});
Deno.test("isUndefined returns false on non null", () => {
  assertEquals(isUndefined(""), false);
  assertEquals(isUndefined(0), false);
  assertEquals(isUndefined([]), false);
  assertEquals(isUndefined({}), false);
  assertEquals(isUndefined(function () {}), false);
  assertEquals(isUndefined(null), false);
});

Deno.test("isNone returns true on null/undefined", () => {
  assertEquals(isNone(null), true);
  assertEquals(isNone(undefined), true);
});
Deno.test("isNone returns false on non null/undefined", () => {
  assertEquals(isNone(""), false);
  assertEquals(isNone(0), false);
  assertEquals(isNone([]), false);
  assertEquals(isNone({}), false);
  assertEquals(isNone(function () {}), false);
});
