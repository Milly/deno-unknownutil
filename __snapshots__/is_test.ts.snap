export const snapshot = {};

snapshot[`isRecordOf<T> > returns properly named function 1`] = `"isRecordOf(isNumber, undefined)"`;

snapshot[`isRecordOf<T> > returns properly named function 2`] = `"isRecordOf((anonymous), undefined)"`;

snapshot[`isUniformTupleOf<T> > returns properly named function 1`] = `"isUniformTupleOf(3, isAny)"`;

snapshot[`isUniformTupleOf<T> > returns properly named function 2`] = `"isUniformTupleOf(3, isNumber)"`;

snapshot[`isUniformTupleOf<T> > returns properly named function 3`] = `"isUniformTupleOf(3, (anonymous))"`;

snapshot[`isInstanceOf<T> > returns properly named function 1`] = `"isInstanceOf(Date)"`;

snapshot[`isInstanceOf<T> > returns properly named function 2`] = `"isInstanceOf((anonymous))"`;

snapshot[`isRecordOf<T, K> > returns properly named function 1`] = `"isRecordOf(isNumber, isString)"`;

snapshot[`isRecordOf<T, K> > returns properly named function 2`] = `"isRecordOf((anonymous), isString)"`;

snapshot[`isRecordObjectOf<T> > returns properly named function 1`] = `"isRecordObjectOf(isNumber, undefined)"`;

snapshot[`isRecordObjectOf<T> > returns properly named function 2`] = `"isRecordObjectOf((anonymous), undefined)"`;

snapshot[`isMapOf<T, K> > returns properly named function 1`] = `"isMapOf(isNumber, isString)"`;

snapshot[`isMapOf<T, K> > returns properly named function 2`] = `"isMapOf((anonymous), isString)"`;

snapshot[`isPartialOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: asOptional(isNumber),
  b: asOptional(isUnionOf([
    isString,
    isUndefined
  ])),
  c: asOptional(isBoolean)
})"
`;

snapshot[`isPartialOf<T> > returns properly named function 2`] = `
"isObjectOf({
  a: asOptional(isNumber),
  b: asOptional(isUnionOf([
    isString,
    isUndefined
  ])),
  c: asOptional(isBoolean)
})"
`;

snapshot[`isMapOf<T> > returns properly named function 1`] = `"isMapOf(isNumber, undefined)"`;

snapshot[`isMapOf<T> > returns properly named function 2`] = `"isMapOf((anonymous), undefined)"`;

snapshot[`isStrictOf<T> > returns properly named function 1`] = `
"isStrictOf(isObjectOf({
  a: isNumber,
  b: isString,
  c: isBoolean
}))"
`;

snapshot[`isStrictOf<T> > returns properly named function 2`] = `"isStrictOf(isObjectOf({a: a}))"`;

snapshot[`isStrictOf<T> > returns properly named function 3`] = `
"isStrictOf(isObjectOf({
  a: isStrictOf(isObjectOf({
    b: isStrictOf(isObjectOf({c: isBoolean}))
  }))
}))"
`;

snapshot[`isLiteralOneOf<T> > returns properly named function 1`] = `'isLiteralOneOf(["hello", "world"])'`;

snapshot[`isParametersOf<T, E> > returns properly named function 1`] = `
"isParametersOf([
  isNumber,
  isString,
  asOptional(isBoolean)
], isArray)"
`;

snapshot[`isParametersOf<T, E> > returns properly named function 2`] = `"isParametersOf([(anonymous)], isArrayOf(isString))"`;

snapshot[`isParametersOf<T, E> > returns properly named function 3`] = `"isParametersOf([], isArrayOf(isString))"`;

snapshot[`isParametersOf<T, E> > returns properly named function 4`] = `
"isParametersOf([
  isParametersOf([
    isParametersOf([
      isNumber,
      isString,
      asOptional(isBoolean)
    ], isArray)
  ], isArray)
])"
`;

snapshot[`isOmitOf<T, K> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  c: isBoolean
})"
`;

snapshot[`isOmitOf<T, K> > returns properly named function 2`] = `"isObjectOf({a: isNumber})"`;

snapshot[`isUnionOf<T> > returns properly named function 1`] = `
"isUnionOf([
  isNumber,
  isString,
  isBoolean
])"
`;

snapshot[`isTupleOf<T, E> > returns properly named function 1`] = `
"isTupleOf([
  isNumber,
  isString,
  isBoolean
], isArray)"
`;

snapshot[`isTupleOf<T, E> > returns properly named function 2`] = `"isTupleOf([(anonymous)], isArrayOf(isString))"`;

snapshot[`isTupleOf<T, E> > returns properly named function 3`] = `
"isTupleOf([
  isTupleOf([
    isTupleOf([
      isNumber,
      isString,
      isBoolean
    ], isArray)
  ], isArray)
])"
`;

snapshot[`isArrayOf<T> > returns properly named function 1`] = `"isArrayOf(isNumber)"`;

snapshot[`isArrayOf<T> > returns properly named function 2`] = `"isArrayOf((anonymous))"`;

snapshot[`isParametersOf<T> > returns properly named function 1`] = `
"isParametersOf([
  isNumber,
  isString,
  asOptional(isBoolean)
])"
`;

snapshot[`isParametersOf<T> > returns properly named function 2`] = `"isParametersOf([(anonymous)])"`;

snapshot[`isParametersOf<T> > returns properly named function 3`] = `"isParametersOf([])"`;

snapshot[`isParametersOf<T> > returns properly named function 4`] = `
"isParametersOf([
  isParametersOf([
    isParametersOf([
      isNumber,
      isString,
      asOptional(isBoolean)
    ])
  ])
])"
`;

snapshot[`isRecordObjectOf<T, K> > returns properly named function 1`] = `"isRecordObjectOf(isNumber, isString)"`;

snapshot[`isRecordObjectOf<T, K> > returns properly named function 2`] = `"isRecordObjectOf((anonymous), isString)"`;

snapshot[`isLiteralOf<T> > returns properly named function 1`] = `'isLiteralOf("hello")'`;

snapshot[`isLiteralOf<T> > returns properly named function 2`] = `"isLiteralOf(100)"`;

snapshot[`isLiteralOf<T> > returns properly named function 3`] = `"isLiteralOf(100n)"`;

snapshot[`isLiteralOf<T> > returns properly named function 4`] = `"isLiteralOf(true)"`;

snapshot[`isLiteralOf<T> > returns properly named function 5`] = `"isLiteralOf(null)"`;

snapshot[`isLiteralOf<T> > returns properly named function 6`] = `"isLiteralOf(undefined)"`;

snapshot[`isLiteralOf<T> > returns properly named function 7`] = `"isLiteralOf(Symbol(asdf))"`;

snapshot[`isRequiredOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  b: isUnionOf([
    isString,
    isUndefined
  ]),
  c: isBoolean
})"
`;

snapshot[`isRequiredOf<T> > returns properly named function 2`] = `
"isObjectOf({
  a: isNumber,
  b: isUnionOf([
    isString,
    isUndefined
  ]),
  c: isBoolean
})"
`;

snapshot[`isIntersectionOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  b: isString
})"
`;

snapshot[`isIntersectionOf<T> > returns properly named function 2`] = `"isString"`;

snapshot[`isIntersectionOf<T> > returns properly named function 3`] = `
"isIntersectionOf([
  isFunction,
  isObjectOf({b: isString})
])"
`;

snapshot[`isObjectOf<T> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  b: isString,
  c: isBoolean
})"
`;

snapshot[`isObjectOf<T> > returns properly named function 2`] = `"isObjectOf({a: a})"`;

snapshot[`isObjectOf<T> > returns properly named function 3`] = `
"isObjectOf({
  a: isObjectOf({
    b: isObjectOf({c: isBoolean})
  })
})"
`;

snapshot[`isPickOf<T, K> > returns properly named function 1`] = `
"isObjectOf({
  a: isNumber,
  c: isBoolean
})"
`;

snapshot[`isPickOf<T, K> > returns properly named function 2`] = `"isObjectOf({a: isNumber})"`;

snapshot[`isTupleOf<T> > returns properly named function 1`] = `
"isTupleOf([
  isNumber,
  isString,
  isBoolean
])"
`;

snapshot[`isTupleOf<T> > returns properly named function 2`] = `"isTupleOf([(anonymous)])"`;

snapshot[`isTupleOf<T> > returns properly named function 3`] = `
"isTupleOf([
  isTupleOf([
    isTupleOf([
      isNumber,
      isString,
      isBoolean
    ])
  ])
])"
`;

snapshot[`isSetOf<T> > returns properly named function 1`] = `"isSetOf(isNumber)"`;

snapshot[`isSetOf<T> > returns properly named function 2`] = `"isSetOf((anonymous))"`;
